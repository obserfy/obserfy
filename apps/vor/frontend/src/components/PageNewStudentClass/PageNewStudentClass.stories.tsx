import { FC } from "react"
import PageNewStudentClass from "./PageNewStudentClass"

export default {
  title: "Core/PageNewStudentClass",
  component: PageNewStudentClass,
  parameters: {
    componentSubtitle: "Just a simple PageNewStudentClass",
  },
}

export const Basic: FC = () => <PageNewStudentClass id="adsfnalskfdb" />
