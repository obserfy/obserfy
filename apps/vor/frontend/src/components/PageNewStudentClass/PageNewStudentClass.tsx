import { FC, useState } from "react"
import { useImmer } from "use-immer"
import { Flex, Button, Box } from "theme-ui"
import { Trans } from "@lingui/macro"
import BackNavigation from "../BackNavigation/BackNavigation"
import { EDIT_STUDENT_CLASS_URL } from "../../routes"

import { useGetStudent } from "../../hooks/api/useGetStudent"
import { Typography } from "../Typography/Typography"
import usePostNewClass from "../../hooks/api/classes/usePostNewClass"
import Input from "../Input/Input"
import Chip from "../Chip/Chip"

import LoadingIndicator from "../LoadingIndicator/LoadingIndicator"
import ErrorMessage from "../ErrorMessage/ErrorMessage"
import { WEEKDAYS } from "../PageNewClass/PageNewClass"
import dayjs from "../../dayjs"
import { navigate } from "../Link/Link"
import usePostStudentClassRelation from "../../hooks/api/students/usePostStudentClassRelation"

interface Props {
  id: string
}
export const PageNewStudentClass: FC<Props> = ({ id }) => {
  const student = useGetStudent(id)

  const [name, setName] = useState("")
  const [startTime, setStartTime] = useState("09:00")
  const [endTime, setEndTime] = useState("10:00")
  const [weekdays, setWeekdays] = useImmer<number[]>([])
  const postNewClass = usePostNewClass()
  const postStudentClassRelation = usePostStudentClassRelation(id)
  const valid = name !== ""

  const handleSubmit = async (): Promise<void> => {
    try {
      const newClassResult = await postNewClass.mutateAsync({
        name,
        weekdays,
        endTime: dayjs(endTime, "HH:mm").toDate(),
        startTime: dayjs(startTime, "HH:mm").toDate(),
      })

      const response = await newClassResult.json()
      await postStudentClassRelation.mutateAsync(response.id)
      await navigate(EDIT_STUDENT_CLASS_URL(id))
    } catch (e) {
      Sentry.captureException(e)
    }
  }

  const isLoading = postNewClass.isLoading || postStudentClassRelation.isLoading

  return (
    <Box mx="auto" sx={{ maxWidth: "maxWidth.sm" }}>
      <BackNavigation to={EDIT_STUDENT_CLASS_URL(id)} text="Edit Class" />
      <Typography.H5 mx={3} mt={3} color="textDisabled">
        {student.data?.name}
      </Typography.H5>
      <Typography.H5 mx={3} mb={3}>
        <Trans>New Class</Trans>
      </Typography.H5>
      <Box m={3}>
        <Input
          label="Name"
          sx={{ width: "100%" }}
          mb={3}
          value={name}
          onChange={(e) => setName(e.target.value)}
        />
        <Flex>
          <Input
            type="time"
            value={startTime}
            onChange={(e) => setStartTime(e.target.value)}
            label="Start Time"
            sx={{ width: "100%" }}
            mb={3}
            mr={3}
          />
          <Input
            type="time"
            value={endTime}
            onChange={(e) => setEndTime(e.target.value)}
            label="End Time"
            sx={{ width: "100%" }}
            mb={3}
          />
        </Flex>
        <Typography.Body mb={2} color="textMediumEmphasis">
          <Trans>Available every</Trans>
        </Typography.Body>
        <Flex
          mb={3}
          sx={{
            flexWrap: "wrap",
          }}
        >
          {WEEKDAYS.map((weekday, i) => (
            <Chip
              mb={2}
              mr={2}
              key={weekday}
              text={weekday}
              activeBackground="primary"
              isActive={weekdays.includes(i)}
              onClick={() =>
                setWeekdays((draft) => {
                  if (draft.includes(i)) {
                    return draft.filter((item) => item !== i)
                  }
                  return [...draft, i]
                })
              }
            />
          ))}
        </Flex>
        <Button
          sx={{ width: "100%" }}
          disabled={!valid || isLoading}
          onClick={handleSubmit}
        >
          {isLoading && <LoadingIndicator mr={2} color="onPrimary" />}
          <Trans>Save</Trans>
        </Button>
        <ErrorMessage error={postNewClass.error} m={3} />
      </Box>
    </Box>
  )
}

export default PageNewStudentClass
