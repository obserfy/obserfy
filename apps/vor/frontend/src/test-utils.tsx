/* eslint-disable import/no-extraneous-dependencies */
import { FC, ReactElement } from "react"
import { QueryClient, QueryClientProvider } from "react-query"
import { ThemeProvider, Box, Theme } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import { I18nProvider } from "@lingui/react"
import { i18n } from "@lingui/core"
import { en } from "make-plural/plurals"
import theme from "./gatsby-plugin-theme-ui"
// @ts-ignore
import enCatalog from "../i18n/lingui/en/messages.js"

// any used here because the catalogs are generated b lingui
// not our responsibility
i18n.loadLocaleData("en", { plurals: en })
i18n.load("en", enCatalog.messages as any)
i18n.activate("en")

const queryClient = new QueryClient()
const history = createHistory(createMemorySource("/"))

const AllProviders: FC = ({ children }) => {
  return (
    <I18nProvider i18n={i18n}>
      <QueryClientProvider client={queryClient}>
        <LocationProvider history={history}>
          <ThemeProvider theme={theme as Theme}>
            <Box sx={{ fontSize: [16, 20] }}>{children}</Box>
          </ThemeProvider>
        </LocationProvider>
      </QueryClientProvider>
    </I18nProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
