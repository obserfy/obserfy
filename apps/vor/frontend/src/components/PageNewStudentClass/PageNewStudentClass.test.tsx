import { render } from "../../test-utils"
import { Basic } from "./PageNewStudentClass.stories"

describe("PageNewStudentClass", () => {
  it("should render correctly", () => {
    const { container } = render(<Basic />)
    expect(container).toMatchSnapshot()
  })
})
