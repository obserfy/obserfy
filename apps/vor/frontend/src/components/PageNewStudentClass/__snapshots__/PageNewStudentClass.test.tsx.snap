// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNewStudentClass should render correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 16px;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    font-size: 20px;
  }
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: auto;
  margin-right: auto;
  max-width: 640px;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: -8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  margin: 16px;
  margin-right: 4px;
  fill: var(--theme-ui-colors-icon);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 24px;
  height: 24px;
  fill: var(--theme-ui-colors-textMediumEmphasis);
}

.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
  margin-bottom: 0;
  color: var(--theme-ui-colors-textMediumEmphasis);
}

.emotion-5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-weight: bold;
  line-height: heading;
  color: var(--theme-ui-colors-text);
  text-rendering: optimizeLegibility;
  font-size: 24px;
  margin-left: 16px;
  margin-right: 16px;
  margin-top: 16px;
  color: var(--theme-ui-colors-textDisabled);
}

.emotion-6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-weight: bold;
  line-height: heading;
  color: var(--theme-ui-colors-text);
  text-rendering: optimizeLegibility;
  font-size: 24px;
  margin-left: 16px;
  margin-right: 16px;
  margin-bottom: 16px;
}

.emotion-7 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin: 16px;
}

.emotion-8 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: var(--theme-ui-colors-textMediumEmphasis);
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-size: 14px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: var(--theme-ui-colors-textMediumEmphasis);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-9 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-bottom: 4px;
}

.emotion-10 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-11 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  width: 100%;
  padding: 8px;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-size: inherit;
  line-height: inherit;
  border: 1px solid;
  border-radius: 4px;
  color: inherit;
  background-color: transparent;
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  width: auto;
  border-color: var(--theme-ui-colors-border);
  background-color: var(--theme-ui-colors-surface);
  font-size: 16px;
  padding: 12px;
  color: var(--theme-ui-colors-text);
  margin-bottom: 16px;
  --theme-ui-input-autofill-bg: var(--theme-ui-colors-background);
  width: 100%;
  width: 100%;
}

.emotion-11:autofill,
.emotion-11:autofill:hover,
.emotion-11:autofill:focus {
  box-shadow: inset 0 0 0 1000px var(--theme-ui-input-autofill-bg);
  font-size: inherit;
}

.emotion-11:autofill:first-line,
.emotion-11:autofill:hover:first-line,
.emotion-11:autofill:focus:first-line {
  font-size: 1rem;
}

.emotion-11:-webkit-autofill,
.emotion-11:-webkit-autofill:hover,
.emotion-11:-webkit-autofill:focus {
  box-shadow: inset 0 0 0 1000px var(--theme-ui-input-autofill-bg);
  font-size: inherit;
}

.emotion-11:-webkit-autofill:first-line,
.emotion-11:-webkit-autofill:hover:first-line,
.emotion-11:-webkit-autofill:focus:first-line {
  font-size: 1rem;
}

.emotion-11:hover,
.emotion-11:focus {
  border-color: var(--theme-ui-colors-primary);
}

.emotion-11::-webkit-input-placeholder {
  font-style: italic;
}

.emotion-11::-moz-placeholder {
  font-style: italic;
}

.emotion-11:-ms-input-placeholder {
  font-style: italic;
}

.emotion-11::placeholder {
  font-style: italic;
}

.emotion-11:disabled {
  opacity: 0.68;
  border-color: var(--theme-ui-colors-border);
  color: var(--theme-ui-colors-textDisabled);
}

.emotion-12 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-16 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  width: 100%;
  padding: 8px;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-size: inherit;
  line-height: inherit;
  border: 1px solid;
  border-radius: 4px;
  color: inherit;
  background-color: transparent;
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  width: auto;
  border-color: var(--theme-ui-colors-border);
  background-color: var(--theme-ui-colors-surface);
  font-size: 16px;
  padding: 12px;
  color: var(--theme-ui-colors-text);
  margin-bottom: 16px;
  margin-right: 16px;
  --theme-ui-input-autofill-bg: var(--theme-ui-colors-background);
  width: 100%;
  width: 100%;
}

.emotion-16:autofill,
.emotion-16:autofill:hover,
.emotion-16:autofill:focus {
  box-shadow: inset 0 0 0 1000px var(--theme-ui-input-autofill-bg);
  font-size: inherit;
}

.emotion-16:autofill:first-line,
.emotion-16:autofill:hover:first-line,
.emotion-16:autofill:focus:first-line {
  font-size: 1rem;
}

.emotion-16:-webkit-autofill,
.emotion-16:-webkit-autofill:hover,
.emotion-16:-webkit-autofill:focus {
  box-shadow: inset 0 0 0 1000px var(--theme-ui-input-autofill-bg);
  font-size: inherit;
}

.emotion-16:-webkit-autofill:first-line,
.emotion-16:-webkit-autofill:hover:first-line,
.emotion-16:-webkit-autofill:focus:first-line {
  font-size: 1rem;
}

.emotion-16:hover,
.emotion-16:focus {
  border-color: var(--theme-ui-colors-primary);
}

.emotion-16::-webkit-input-placeholder {
  font-style: italic;
}

.emotion-16::-moz-placeholder {
  font-style: italic;
}

.emotion-16:-ms-input-placeholder {
  font-style: italic;
}

.emotion-16::placeholder {
  font-style: italic;
}

.emotion-16:disabled {
  opacity: 0.68;
  border-color: var(--theme-ui-colors-border);
  color: var(--theme-ui-colors-textDisabled);
}

.emotion-21 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
  margin-bottom: 8px;
  color: var(--theme-ui-colors-textMediumEmphasis);
}

.emotion-22 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-bottom: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.emotion-23 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-bottom: 8px;
  margin-right: 8px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  background-color: var(--theme-ui-colors-surface);
  display: inline-block;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  border-radius: 6px;
  border-color: var(--theme-ui-colors-border);
  border-width: 1px;
  border-style: solid;
  white-space: nowrap;
}

.emotion-23:hover {
  background-color: var(--theme-ui-colors-primaryLightest);
}

.emotion-24 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
}

.emotion-37 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary);
  border: 0;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  box-sizing: border-box;
  font-size: 14px;
  color: var(--theme-ui-colors-onPrimary);
  background-color: var(--theme-ui-colors-primary);
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  touch-action: manipulation;
  box-shadow: rgba(27, 31, 35, 0.1) 0px 1px 0px 0px,rgba(255, 255, 255, 0.03) 0px 1px 0px 0px inset;
  border-color: var(--theme-ui-colors-border);
  border-style: solid;
  border-width: 1px;
  line-height: 1;
  width: 100%;
}

.emotion-37:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-37:hover,
.emotion-37:focus {
  background-color: var(--theme-ui-colors-primaryDark);
}

.emotion-37:hover:disabled,
.emotion-37:focus:disabled {
  background-color: var(--theme-ui-colors-primary);
}

.emotion-38 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
  margin: 16px;
  color: var(--theme-ui-colors-error);
  text-align: center;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <a
        href="/dashboard/students/profile/classes/edit?studentId=adsfnalskfdb"
        style="display: block;"
      >
        <div
          class="emotion-2"
        >
          <div
            class="emotion-3"
          />
          <p
            class="emotion-4"
          >
            Edit Class
          </p>
        </div>
      </a>
      <h5
        class="emotion-5"
      />
      <h5
        class="emotion-6"
      >
        New Class
      </h5>
      <div
        class="emotion-7"
      >
        <label
          class="emotion-8"
        >
          <div
            class="emotion-9"
          >
            Name
          </div>
          <div
            class="emotion-10"
          >
            <input
              class="emotion-11"
              value=""
            />
          </div>
        </label>
        <div
          class="emotion-12"
        >
          <label
            class="emotion-8"
          >
            <div
              class="emotion-9"
            >
              Start Time
            </div>
            <div
              class="emotion-10"
            >
              <input
                class="emotion-16"
                type="time"
                value="09:00"
              />
            </div>
          </label>
          <label
            class="emotion-8"
          >
            <div
              class="emotion-9"
            >
              End Time
            </div>
            <div
              class="emotion-10"
            >
              <input
                class="emotion-11"
                type="time"
                value="10:00"
              />
            </div>
          </label>
        </div>
        <p
          class="emotion-21"
        >
          Available every
        </p>
        <div
          class="emotion-22"
        >
          <div
            class="emotion-23"
          >
            <p
              class="emotion-24"
            >
              Sunday
            </p>
          </div>
          <div
            class="emotion-23"
          >
            <p
              class="emotion-24"
            >
              Monday
            </p>
          </div>
          <div
            class="emotion-23"
          >
            <p
              class="emotion-24"
            >
              Tuesday
            </p>
          </div>
          <div
            class="emotion-23"
          >
            <p
              class="emotion-24"
            >
              Wednesday
            </p>
          </div>
          <div
            class="emotion-23"
          >
            <p
              class="emotion-24"
            >
              Thursday
            </p>
          </div>
          <div
            class="emotion-23"
          >
            <p
              class="emotion-24"
            >
              Friday
            </p>
          </div>
          <div
            class="emotion-23"
          >
            <p
              class="emotion-24"
            >
              Saturday
            </p>
          </div>
        </div>
        <button
          class="emotion-37"
          disabled=""
        >
          Save
        </button>
        <p
          class="emotion-38"
        />
      </div>
    </div>
  </div>
</div>
`;
